mm/tlsf/mm_tlsf.c

```diff
@@ -1005,6 +1004,13 @@ FAR void *mm_malloc(FAR struct mm_heap_s *heap, size_t size)
 {
   FAR void *ret;
 
+  /* In case of zero-length allocations allocate the minimum size object */
+
+  if (size < 1)
+    {
+      size = 1;
+    }
+
```

Comment:
Passing size=0 and size=1 to mm_malloc have the same effect. A minimum chunk size is allocated. So I think the above can be removed.